MelodyGenerator.js

```diff
@@ -0,0 +1,145 @@
+const durations = require('./durations.js');
+const MelodyQueue = require('./MelodyQueue.js');
+const waveGenerator = require('./WaveGenerator.js');
+const note = require('./note.js');
+
+//Global variables that are accessible to all
+let melody = new MelodyQueue();
+let macros = new Map(); // In case a macro needs to be defined
+const regex1 = /repeat (\d) times (.+)/; // Regex for repeat commands
+const regex2 = /((note\(|pause\()((.{2,3})(;(\w\w))*)(,\s?([A-Z]))?\))/; //Regex for notes, pauses
+const regex3 = /define (\w+) is (.+)/;  //Regex for defining macros
+const regex4 = /(play )?(.+)/; // Regex for playing non-repeating macros
+
+class MelodyGenerator
+{
+
+	parseMacro(lineString) {
+		// define mel01 is pause(HT) | note(E5, E)
+		const parsedMacro = regex3.exec(lineString); 
+		const macroName = parsedMacro[1];
+		const line = parsedMacro[2].split("|");
```

Comment:
Errr... sort of. :-D  I called it "line" probably because it took the lineString and broke it into an array, each index holding an element of sound. 

I think what you're trying to tell me is that it's not very intuitive for others reading the code. Right? This is a very good point.

Reply:
Yes, `line` is not a intuitive variable - `line` almost always is a string. But If you was thinking about variable more than 2 hours then `line` is a good option :-D